plugins/async-node/ios/Tests/AsynNodePluginTests.swift

```diff
@@ -520,6 +520,273 @@ class AsyncNodePluginTests: XCTestCase {
         XCTAssertEqual(expectedMultiNode3Text, "test")
         XCTAssertEqual(expectedMultiNode4Text, "undefined")
     }
+    
+    func testChatMessageReplaceAsyncNodeWithProvidedNode() {
+       let handlerExpectation = XCTestExpectation(description: "first data did not change")
+       
+       let context = JSContext()
+       var count = 0
+       
+       let resolve: AsyncHookHandler = { _,_ in
+           handlerExpectation.fulfill()
+           
+           if count == 1 {
+               return .singleNode(ReplacementNode.encodable(
+                   AssetPlaceholderNode(asset: PlaceholderNode(id: "text", type: "text", value: "new node"))
+               ))
+           }
+           
+           return .singleNode(ReplacementNode.concrete(context?.evaluateScript("") ?? JSValue()))
+       }
+       
+       let asyncNodePluginPlugin = AsyncNodePluginPlugin()
+       let plugin = AsyncNodePlugin(plugins: [asyncNodePluginPlugin], resolve)
```

Comment:
out of scope for the PR, but this constructor API is inconsistent across platforms